### PR TITLE
Big clean in dependency management

### PR DESCRIPTION
**Pull request by enthuan**
Bumped versions of Gradle, AGP, Kotlin and a few more.

Updated Build logic module.

Removed Accompanist and migrated to native alternatives.

Removed unused libraries

### CODE REVIEW

**Comment by martinbonnin** [comment]
> Removed Build logic module. I plan to add a full KMP support and this module was not what I was used to.

I'd push back on this. Being able to factor build logic code is going to be handy. Ping me on android-dev-fr and we can do a small pairing session on it if you feel like it. Overall `build-logic` is a relatively accepted pattern these days (see [nowindanroid](https://github.com/android/nowinandroid/tree/main/build-logic) for an example and https://developer.squareup.com/blog/herding-elephants/ for an intro to convention plugin). They are very due for an update but I'd like to keep the `build-logic` folder if possible.

**Comment by enthuan** [comment]
After reading the documentation you provided, I understand more the goal of the `build-logic` module. I split the convention plugins in several parts : 
- `AndroidApplication` for the basics on an android app
- `AndroidCompose` for setting the dependencies, build feature and compiler version
- `MultiplatformLibrary` for setting the java version and applying the standard kmp plugins. For the moment, only the androidTarget is added.
- `SigningConfig` for defining the signing configuration of an android application.

Please do not hesitate to give me a feedback on it, specially if you prefer more/less granularity on those plugins.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Wed Apr 20 15:34:11 CEST 2022
+#Mon Feb 12 16:24:49 CET 2024
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1-bin.zip
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-bin.zip
```
Comment (martinbonnin) [inline gradle/wrapper/gradle-wrapper.properties:4]: Make it 8.6?

**Review by martinbonnin — APPROVED** [review]
Very cool, thanks! A minor comment about Gradle version but otherwise 👌

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (martinbonnin) [reply gradle/wrapper/gradle-wrapper.properties:4]: Merging as is, we can always bump lter.

**Pull request merged**